### PR TITLE
Use `Params` in script score

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we now use `params` in the script score to reduce the number of compiling

* **What is the current behavior?** (You can also link to an open issue here)
Currently, we hard code the weights in the painless script, which means OpenSearch will have to recompile the painless script whenever the field name or the weight is changed. 

* **What is the new behavior (if this is a feature change)?**
We put the weights in params and use the params in the script. So Opensearch can reuse the compiled scripts by just passing params.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
running

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
no

* **Other information**:
https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-using.html#prefer-params

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/tensor_search/models/score_modifiers_object.py [hunk 1]
```diff
@@ -74,23 +79,42 @@ def __init__(self, **data):
...
         return v
 
-    def to_painless_script(self) -> str:
-        """Convert this ScoreModifier to a painless script to modify the score."""
-        mult = [x.to_painless_script("multiply_score_by") for x in self.multiply_score_by] if self.multiply_score_by is not None else []
-        add = [x.to_painless_script("add_to_score") for x in self.add_to_score] if self.add_to_score is not None else []
+    def to_script_score(self) -> dict:
+        """Convert this ScoreModifier to a dictionary to modify the score.
+        returns:
+            A dictionary that can be used as a script_score in an Elasticsearch query.
```
Comment (farshidz) [inline src/marqo/tensor_search/models/score_modifiers_object.py]: OpenSearch

**Review by farshidz — APPROVED** [review]

File: tests/tensor_search/test_score_modifiers_search.py
Comment (pandu-k) [inline tests/tensor_search/test_score_modifiers_search.py:1]: Thanks Li! Can you please add more tests about the scripts and param dicts generated by score modifiers, including edge cases?
